### PR TITLE
Task

### PR DESCRIPTION
**Pull request by alexbord**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,7 +1,23 @@
 'use strict';
 
+const path = require('path');
 const app = require('express')();
 const tasksContainer = require('./tasks.json');
+const webpack = require('webpack');
```
Comment (asci) [inline server.js]: is it possible to use here `webpack-dev-server`?  in this case, the server will be more focused on API

**Review by asci — APPROVED** [review]
In general, I like it. There are small style issues, but it really matters of conventions

File: src/components/Form.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      title,
+      description
+    };
+
+    this.handleInputChange = this.handleInputChange.bind(this);
+  }
+
+  handleInputChange(e) {
+    const { value, name } = e.target;
+    this.setState({
```
Comment (ghost) [inline src/components/Form.js:25]: Why do you mutate the state instead of the props here?

**Comment by reaktivo** [comment]
There's a few subtle bugs when trying to delete an out-of-range task id, for example, `DELETE /task/delete/999999`. This same bug applies to `GET` can you fix them?

Also, can you update the API into a more be more restful, to have a single endpoint that handles different http methods?

File: server.js [hunk 3]
```diff
@@ -90,14 +108,16 @@ app.put('/task/update/:id/:title/:description', (req, res) => {
  */
 app.post('/task/create/:title/:description', (req, res) => {
   const task = {
-    id: tasksContainer.tasks.length,
+    id: lastTaskId,
```
Comment (ghost) [inline server.js]: Maybe  a timestamp is better approach here

File: server.js [hunk 4]
```diff
@@ -116,16 +136,15 @@ app.delete('/task/delete/:id', (req, res) => {
   const id = parseInt(req.params.id, 10);
 
   if (!Number.isNaN(id)) {
-    const task = tasksContainer.tasks.find(item => item.id === id);
+    const taskIndex = tasksContainer.tasks.findIndex(item => item.id === id);
   
-    if (task !== null) {
-      const taskIndex = tasksContainer.tasks;
+    if (taskIndex !== null) {
```
Comment (ghost) [inline server.js]: index will be -1 not nil

File: src/components/Item.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+    this.state = {
+      mode: 'view'
+    };
+
+    this.onEditClick = this.onEditClick.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { isDoing } = this.props;
```
Comment (ghost) [inline src/components/Item.js:21]: Can be more simple than this

File: src/containers/App.js [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+    if (id) {
+      this.props.updateTask(id, title, description);
+    } else {
+      this.props.createTask(title, description);
+    }
+  }
+
+  onItemEditClick(id) {
+
```
Comment (ghost) [inline src/containers/App.js]: Empty :)

File: src/index.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Root from './containers/Root';
+import { configureStore } from './store/configureStore';
+
+import './styles/main.scss';
```
Comment (ghost) [inline src/index.js:6]: Nice to see style import here

File: src/styles/main.scss [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+.task-form {
+  display: flex;
+  justify-content: space-around;
+  margin: 15px 0 0;
+
+  .task-form-inputs {
+    flex-grow: 1;
+  }
+
+  & > div {
```
Comment (ghost) [inline src/styles/main.scss]: Nesting >

File: server.js [hunk 9]
```diff
@@ -1,19 +1,23 @@
...
 const path = require('path');
+const bodyParser = require('body-parser');
 const app = require('express')();
 const tasksContainer = require('./tasks.json');
 const webpack = require('webpack');
 const config = require('./webpack.config.dev');
 const compiler = webpack(config);
 let lastTaskId = tasksContainer.tasks.length + 1;
 
+const findTask = id => tasksContainer.tasks.find(item => item.id === id) || null;
```
Comment (ghost) [inline server.js]: the `find` method returns `null` if no items match your condition. Why did you add `|| null` at the end of your function?

Comment (alexbord) [reply server.js]: https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/find#Return_value -> `undefined`

**Review by reaktivo — APPROVED** [review]

**Review by ghost — APPROVED** [review]
